### PR TITLE
Functionize custom_action() sections for readability

### PR DESCRIPTION
**Pull request by kgaillot**
custom_action() has always been long and difficult to follow, so hopefully separating it into smaller pieces will help

### CODE REVIEW

**Comment by kgaillot** [comment]
@clumens ,  can you review this when you have time? Low priority

**Comment by kgaillot** [comment]
Take a look at the function before reviewing this, so you can see what I mean by difficult to follow. :) Let me know if you think splitting/naming anything differently would be easier to follow.

File: lib/pengine/utils.c [hunk 1]
```diff
@@ -447,6 +447,79 @@ find_existing_action(const char *key, pe_resource_t *rsc, pe_node_t *node,
...
+new_action(char *key, const char *task, pe_resource_t *rsc, pe_node_t *node,
+           bool optional, bool for_graph, pe_working_set_t *data_set)
+{
+    pe_action_t *action = calloc(1, sizeof(pe_action_t));
+
+    CRM_ASSERT(action != NULL);
+
+    action->rsc = rsc;
+    action->task = strdup(task); CRM_ASSERT(action->task != NULL);
+    action->uuid = key;
```
Comment (clumens) [inline lib/pengine/utils.c:494]: Did you mean to strip the strdup off this line?

**Comment by clumens** [comment]
This all looks good to me.  It's much easier to follow now.

File: lib/pengine/utils.c (hunk 1 above)
Comment (kgaillot) [reply lib/pengine/utils.c:494]: Yes, that's what the commit message is referring to ... previously custom_action() would strdup() the provided key if this block were hit, and always free the provided key. Now, we use the provided key here directly and only free it if this isn't hit.

**Pull request merged**